src/plugins/event_annotation_listing/public/components/group_editor_flyout/group_editor_flyout.tsx

```diff
@@ -0,0 +1,254 @@
+  queryInputServices,
+  searchSessionId,
+  refreshSearchSession,
+  timePickerQuickRanges,
+}: {
+  group: EventAnnotationGroupConfig;
+  updateGroup: (newGroup: EventAnnotationGroupConfig) => void;
+  onClose: () => void;
+  onSave: () => void;
+  savedObjectsTagging: SavedObjectsTaggingApi;
+  dataViews: DataView[];
+  createDataView: (spec: DataViewSpec) => Promise<DataView>;
+  LensEmbeddableComponent: LensEmbeddableComponent;
+  queryInputServices: QueryInputServices;
+  searchSessionId: string;
+  refreshSearchSession: () => void;
+  timePickerQuickRanges: Array<{ from: string; to: string; display: string }> | undefined;
+}) => {
+  const flyoutHeadingId = useMemo(() => htmlIdGenerator()(), []);
+  const flyoutBodyOverflowRef = useRef<Element | null>(null);
+  useEffect(() => {
+    if (!flyoutBodyOverflowRef.current) {
+      flyoutBodyOverflowRef.current = document.querySelector('.euiFlyoutBody__overflow');
+    }
+  }, []);
+
+  const [hasAttemptedSave, setHasAttemptedSave] = useState(false);
+
+  const resetContentScroll = useCallback(
+    () => flyoutBodyOverflowRef.current && flyoutBodyOverflowRef.current.scroll(0, 0),
+    []
+  );
+
+  // save the spec for the life of the component since the user might change their mind after selecting another data view
+  const [adHocDataView, setAdHocDataView] = useState<DataView>();
+
+  useEffect(() => {
+    if (group.dataViewSpec) {
+      createDataView(group.dataViewSpec).then(setAdHocDataView);
+    }
+  }, [createDataView, group.dataViewSpec]);
+
+  const dataViews = useMemo(() => {
+    const items = [...globalDataViews];
+    if (adHocDataView) {
+      items.push(adHocDataView);
+    }
+    return items;
+  }, [adHocDataView, globalDataViews]);
+
+  const [selectedAnnotation, _setSelectedAnnotation] = useState<EventAnnotationConfig>();
+  const setSelectedAnnotation = useCallback(
+    (newValue: EventAnnotationConfig | undefined) => {
+      if ((!newValue && selectedAnnotation) || (newValue && !selectedAnnotation))
+        resetContentScroll();
+      _setSelectedAnnotation(newValue);
+    },
+    [resetContentScroll, selectedAnnotation]
+  );
+  const onClose = () => (selectedAnnotation ? setSelectedAnnotation(undefined) : parentOnClose());
+
+  const showPreview = !useIsWithinBreakpoints(['xs', 's']);
+
+  return (
+    <EuiFlyout
+      onClose={onClose}
+      paddingSize="m"
+      size="l"
+      hideCloseButton
+      outsideClickCloses={false}
+    >
+      <EuiFlexGroup
+        css={css`
+          height: 100%;
+          overflow-y: auto;
+        `}
+        gutterSize="none"
+      >
+        <EuiFlexItem
+          grow={false}
+          css={css`
+            ${showPreview ? 'width: 360px;' : ''}
+            border-right: 1px solid ${euiThemeVars.euiColorLightShade};
+          `}
+        >
+          <EuiFlyoutHeader hasBorder aria-labelledby={flyoutHeadingId}>
+            <EuiTitle size="xs">
+              <h2 id={flyoutHeadingId}>
```

Comment:
Again, a subjective decision, but would it be better to wrap this `EuiTitle` component and `h2` element around the title text only (rather than the title and the back button)?

Reply:
I see where you're coming from, but I like it this way since we don't have to duplicate the elements between the two titles in the markup